boutdata/tests/test_collect.py

```diff
@@ -0,0 +1,2773 @@
+    collect_kwargs,
+    squash_kwargs={},
+):
+    # Apply effect of arguments to expected data
+    if not collect_kwargs["xguards"]:
+        remove_xboundaries(expected, expected["MXG"])
+    if collect_kwargs["yguards"] is True and doublenull:
+        remove_yboundaries_upper_divertor(
+            expected, expected["MYG"], expected["ny_inner"]
+        )
+    if not collect_kwargs["yguards"]:
+        remove_yboundaries(expected, expected["MYG"], expected["ny_inner"], doublenull)
+
+    collect_kwargs = collect_kwargs.copy()
+    if squash:
+        squashoutput(path, outputname="boutdata.nc", **collect_kwargs)
+        collect_kwargs["prefix"] = "boutdata"
+        # Delete dump files to be sure we do not read from them
+        dump_names = glob(str(path.joinpath("BOUT.dmp.*.nc")))
+        for x in dump_names:
+            Path(x).unlink()
+        # Reset arguments that are taken care of by squashoutput
+        for x in ("tind", "xind", "yind", "zind"):
+            if x in collect_kwargs:
+                collect_kwargs.pop(x)
+        # Never remove x-boundaries when collecting from a squashed file without them
+        collect_kwargs["xguards"] = True
+        # Never remove y-boundaries when collecting from a squashed file without them
+        collect_kwargs["yguards"] = "include_upper"
+
+    for varname in expected:
+        actual = collect(varname, path=path, **collect_kwargs)
+        npt.assert_array_equal(expected[varname], actual)
+        actual_keys = list(actual.attributes.keys())
+        if varname in expected_attributes:
+            for a in expected_attributes[varname]:
+                assert actual.attributes[a] == expected_attributes[varname][a]
+                actual_keys.remove(a)
+
+        if "fieldperp" in varname:
+            assert actual.attributes["yindex_global"] == fieldperp_global_yind
+            actual_keys.remove("yindex_global")
+
+        assert actual_keys == ["bout_type"]
+
+        if "field3d_t" in varname:
+            assert actual.attributes["bout_type"] == "Field3D_t"
+        elif "field3d" in varname:
+            assert actual.attributes["bout_type"] == "Field3D"
+        elif "field2d_t" in varname:
+            assert actual.attributes["bout_type"] == "Field2D_t"
+        elif "field2d" in varname:
+            assert actual.attributes["bout_type"] == "Field2D"
+        elif "fieldperp_t" in varname:
+            assert actual.attributes["bout_type"] == "FieldPerp_t"
+        elif "fieldperp" in varname:
+            assert actual.attributes["bout_type"] == "FieldPerp"
+        elif "_t" in varname or varname == "t_array":
+            assert actual.attributes["bout_type"] == "scalar_t"
+        else:
+            assert actual.attributes["bout_type"] == "scalar"
+
+
+class TestCollect:
+    @pytest.mark.parametrize("squash", [False, True])
+    @pytest.mark.parametrize("collect_kwargs", collect_kwargs_list)
+    def test_core_min_files(self, tmp_path, squash, collect_kwargs):
```

Comment:
Although the function names are quite descriptive, I've found simple one-liners describing the tests very helpful in other projects